Standards/scs-0111-v1-volume-type-decisions.md

```diff
@@ -0,0 +1,90 @@
+
+This first decision will have impacts on upstream OpenStack development, as those things, that would be nice to discover, may not be currently dicoverable by users or not at all.
+
+There are severel aspects of volume types, which will be discussed in the following:
+
+### Options considered
+
+#### Encryption
+
+Encryption for volumes is an option which has to be configured within the volume type. As an admin it is possible to set encryption-provider, key size, cipher and control location. As an admin it is also currently possible to see these configurations in a volume type with list and show commands. A user should not see these parameters in detail, but a boolean value that descibes whether encryption is used or not. Currently this is not possible in upstream OpenStack.
+
+**Conclusion**: This is a solid aspect to be standardized. But it will need work on OpenStack, to have a boolean value presented to the users.
+
+#### Backend Name
+
+OpenStack Cinder works with a lot of different backends. They all have some kind of special features, which might be attractive for a user. But showing the nam eof the backend to users is also considered a security risk by Cinder developers. Overall it is always an option to make users aware of special features through the name and description of a volume type and sometimes even through extra_specs.
+
+**Conclusion**: This should not be standardized.
+
+#### Availability Zones
+
+Availability Zones are used in Nova and Cinder seperatly to provide an often also physical separation of compute hosts or storage nodes. This leads to two options to consider:
+
+1. Multiple Volume AZs: This might be used if there are different backends present in one IaaS structure. The different volume types are usually used for the different volume AZs. This makes migration between those AZs only be possible for administrators.
+
+2. Volume Types that can be attached to multiple Nova Azs: This option can be seen in the extra specs of a volume type also by normal users. Another option is to use backend specific options, as for example with ceph that directly interacts with nova for this. In that case there will not be any visible extra specs for the users.
+
+Another question is how many providers use one of these options or both.
+
+**Conclusion**: The first part doesn't make much sense to standardize, as migration between the volume types can only be done by admins. However the second part might be noteable, but due to the variety of configuration options very hard to standardize.
+
+#### Multiattach
+
+It is possible in a few backends to attach a volume to multiple VMs. This hat to be configured in the Volume Type and this information is also accessable for users. Nevertheless this option also needs a lot of work from users, as those types of volumes have to have a file system, that is capable of multiattach. Many providers don't provide multiattach.
```

Comment:
What about Cross Attach AZ?

Reply:
What excactly do you mean with cross AZ, especially in this part? I tried to describe, what is possible to configure and what is possible to see for users. This does not match excactly. And we have two types of AZ that to me are quite different. A) Volumes can reside in different Volume AZs e.g. mapped to different backends. B) Volumes can be attached to VMs in specific AZs, that is configured for Ceph in Ceph and Nova.